ark-core/src/proof_of_funds.rs

```diff
@@ -66,6 +71,39 @@ impl Input {
             pk,
             spend_info,
             is_onchain,
+            preimage: None,
+        }
+    }
+}
+
+impl From<&ArkNote> for Input {
+    fn from(value: &ArkNote) -> Self {
+        let spending_info = value.vtxo_script().spend_info();
+
+        // this is inside the taproot script path
+        let node_script = value.note_script.clone();
+        let Some(control_block) =
+            spending_info.control_block(&(node_script.clone(), LeafVersion::TapScript))
+        else {
+            // FIXME: probably we need a tryfrom?
+            panic!("no control block found");
+        };
+
```

Comment:
I think we need to model things differently so that the conversion from Ark note to proof of funds input is infallible. An Ark note always has the same structure AFAIK, so it should just work.

Reply:
Ok to me this just sounds calling `.unwrap` or `expect` on the `control_block `?